pkg/scheduler/internal/cache/snapshot.go

```diff
@@ -138,28 +138,28 @@ func (s *Snapshot) NumNodes() int {
 	return len(s.nodeInfoList)
 }
 
-type podLister []*framework.NodeInfo
+type podInfoLister []*framework.NodeInfo
 
-// List returns the list of pods in the snapshot.
-func (p podLister) List(selector labels.Selector) ([]*v1.Pod, error) {
-	alwaysTrue := func(*v1.Pod) bool { return true }
+// List returns the list of PodInfos in the snapshot.
+func (p podInfoLister) List(selector labels.Selector) ([]*framework.PodInfo, error) {
+	alwaysTrue := func(*framework.PodInfo) bool { return true }
 	return p.FilteredList(alwaysTrue, selector)
 }
 
 // FilteredList returns a filtered list of pods in the snapshot.
-func (p podLister) FilteredList(filter framework.PodFilter, selector labels.Selector) ([]*v1.Pod, error) {
+func (p podInfoLister) FilteredList(filter framework.PodInfoFilter, selector labels.Selector) ([]*framework.PodInfo, error) {
 	// podFilter is expected to return true for most or all of the pods. We
 	// can avoid expensive array growth without wasting too much memory by
 	// pre-allocating capacity.
 	maxSize := 0
 	for _, n := range p {
 		maxSize += len(n.Pods)
 	}
-	pods := make([]*v1.Pod, 0, maxSize)
+	pods := make([]*framework.PodInfo, 0, maxSize)
 	for _, n := range p {
 		for _, p := range n.Pods {
```

Comment:
nit: Can we rename `NodeInfo.Pods` to `NodeInfo.PodInfos`? then use n.PodInfos here.

Reply:
this will need to be done in a separate PR, it is a not small change because nodeinfo.pods is used in many places.